moveit_ros/moveit_servo/src/servo_parameters.cpp

```diff
@@ -64,16 +64,20 @@ void declareOrGetParam(T& output_value, const std::string& param_name, const rcl
   }
   catch (const rclcpp::exceptions::InvalidParameterTypeException& e)
   {
-    RCLCPP_WARN_STREAM(logger, "InvalidParameterTypeException(" << param_name << "): " << e.what());
-
-    // Catch a <double> parameter written in the yaml as "1" being considered an <int>
-    if (std::is_same<T, double>::value)
+    // Ignore int's when a double is expected because int's get cast properly
+    int int_parameter;
+    if (node->get_parameter<int>(param_name, int_parameter) && std::is_floating_point<T>::value)
     {
-      node->undeclare_parameter(param_name);
-      output_value = node->declare_parameter<int>(param_name, 0);
+      if (std::is_same<T, double>::value)
+      {
+        node->undeclare_parameter(param_name);
+        // Implicitly cast to double
+        output_value = node->declare_parameter<int>(param_name, 0);
```

Comment:
OK, I'm fine with that. That will force people to quickly ensure their yaml files are correct   ;)

Reply:
Done. May this be the last push I make on this silly PR   ;)

Reply:
That's exactly what I pointed out above @tylerjw , thanks for confirming.

My question is whether it's possible, with the current rclcpp interfaces, to get the specified integer parameter, but still declare the parameter as double. If that's not possible, it's a shortcoming of rclcpp because user-configurations should not define/restrict a node's parameter types.
If it's possible, it would be a good idea to provide that functionality with rosparam_shortcuts (and use it here among other places).

Reply:
I don't understand how to accept integers but still declare the parameter as a double with the current system in ROS2.  I think I now understand what you were saying up above.  I'm sorry it took me a while to get here.

I hope I'm just missing something and that it is possible to still declare the parameter as a double and up-convert any integer that comes in.  If that's not possible maybe we could find a way to contribute it upstream (a new "number" type maybe).

Reply:
Sorry I missed your comment about dynamic parameters at first, too